Fix flocculation plant deforming when it has a cover emitting a strong redstone signal

NotAPenguin0 opened the pull request:
As it turns out, cofhcore turns any redstone-powered water into a  `BlockTickingWater`, making it fail the structure check against regular water blocks. This PR fixes it by adding a new `ofAnyWater` structure element that also tests this class.

Alexdoru reviewed (commented):
Is cofh a hard dep on GT5u ? In the multiblock fishing pong they use a reflection check to see if the mod is present. You might need to check if the mod is loaded before doing that. And once that is done would it be possible to use your same mechanic with the fishing pond multi and another one that has the same issue. Basically all multis that use a water block have the same issue

NotAPenguin0 commented:
Seems to be a compileOnly dep, so it would need a modloaded check yes

NotAPenguin0 commented:
- Guarded the class usages against a modloaded check
- Added a check for distilled water since the chemical bath accepts it
- Fixed the chemical bath counting distilled water as two blocks for some reason
- Added checks for cofhcore water classes to chemical bath and fishing pond

serenibyss commented on src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/MTEIndustrialWashPlant.java:
```diff
@@ -146,13 +149,7 @@ public IStructureDefinition<MTEIndustrialWashPlant> getStructureDefinition() {
-                    ofChain(
-                        isAir(),
-                        ofBlockAnyMeta(Blocks.water),
-                        ofBlockAnyMeta(Blocks.flowing_water),
```
Flowing water here seems to no longer be checked, is this intentional?

NotAPenguin0 replied:
It's not, oops
